### PR TITLE
put vale in default location to have tools 'just work'

### PR DESCRIPTION
**Pull request by maxandersen**
by putting .vale/vale.ini at .vale.ini then cli and ide integration "just works".

Any reason why not having it this way instead of having to pass the config in at every location/ide config?

### CODE REVIEW

**Comment by ebullient** [comment]
I was trying to keep all configuration needed for docs contained within the docs directory so we don't have a plethora of .whatever files running everywhere.

I don't expect most/many devs to install vale, so I did not optimize for that path.

**Comment by maxandersen** [comment]
updated the references for `vale.ini` that I could find.

ready for review/merge assuming the testing passes.

**Comment by ebullient** [comment]
> updated the references for `vale.ini` that I could find.
> 
> 
> 
> ready for review/merge assuming the testing passes.

The -Dvale cases must be tested manually.. only used for running locally when you don't have vale set up. Container paths for test containers may prove interesting

**Comment by maxandersen** [comment]
> I also only see one file change? (I am traveling for a wedding.. can look more when I am home)

whoops - missed a push. now pushed.

**Comment by maxandersen** [comment]
> The -Dvale cases must be tested manually.. only used for running locally when you don't have vale set up. Container paths for test containers may prove interesting

so I see that the test assumes everything is in two dirs (vale and asciidoc)  - should be easy enough to mount .vale.ini too but it seem to be configured using absolute argument positions thus not sure how much of this I can just change?

File: docs/src/main/asciidoc/doc-contribute-docs-howto.adoc [hunk 1]
```diff
@@ -90,27 +88,31 @@ Include errors in the results.
...
+If you install the https://vale.sh/docs/vale-cli/installation/[install the Vale CLI], you can run the checks from the `docs` directory:
 
 [source,shell]
 ----
-vale --config=docs/.vale/vale.ini --minAlertLevel=warning docs/src/main/asciidoc
+cd docs
+vale --minAlertLevel=warning src/main/asciidoc
 ----
 
+If you want to run it from another directory use `--config=/path/to/quarks/docs/.vale.ini`.
```
Comment (ebullient) [inline docs/src/main/asciidoc/doc-contribute-docs-howto.adoc:99]:
```suggestion
If you want to run it from another directory use `--config=/path/to/quarkus/docs/.vale.ini`.
```

File: docs/src/main/asciidoc/doc-contribute-docs-howto.adoc [hunk 2]
```diff
@@ -90,27 +88,31 @@ Include errors in the results.
...
+
 See the https://vale.sh/manual/[Vale CLI Manual] for details.
 
 === Vale IDE plugins
 
 https://vale.sh/docs/integrations/guide/[Vale IDE integrations] rely on installation the Vale CLI.
 
-Each has it's own configuration requirements. The Visual Studio Code IDE extension, for example, requires two workspace settings:
+Each has it's own configuration requirements. The Visual Studio Code IDE extension, for example, requires you to tell where valeCLI is installed in workspace settings:
```
Comment (ebullient) [inline docs/src/main/asciidoc/doc-contribute-docs-howto.adoc:107]:
```suggestion
Each has it's own configuration requirements. The Visual Studio Code IDE extension, for example, requires you to tell where the vale CLI is installed in workspace settings:
```

File: docs/src/main/asciidoc/doc-contribute-docs-howto.adoc [hunk 3]
```diff
@@ -90,27 +88,31 @@ Include errors in the results.
...
 
 [source,json]
 ----
-"vale.valeCLI.config": "/path/to/quarkus/docs/.vale/vale.ini",
 "vale.valeCLI.path": "/path/to/vale"
 ----
 
+The plugin will pick up the `.vale.ini` settings automatically relatively to the documentation files.
```
Comment (ebullient) [inline docs/src/main/asciidoc/doc-contribute-docs-howto.adoc:114]:
```suggestion
The plugin will discover the `.vale.ini` configuration automatically.
```

File: docs/src/test/java/io/quarkus/docs/vale/ValeAsciidocLint.java [hunk 4]
```diff
@@ -154,7 +154,10 @@ public Map<String, ChecksBySeverity> lintFiles() throws Exception {
 
         DockerImageName valeImage = DockerImageName.parse(imageName);
 
-        List<String> command = new ArrayList<>(List.of("--config=/vale/vale.ini",
+        // --config=vale.ini is not really necessary as it is the default
+        // location, but using it here explicitly to have the test case fail
+        // if the file is missing.
+        List<String> command = new ArrayList<>(List.of("--config=vale.ini",
```
Comment (ebullient) [inline docs/src/test/java/io/quarkus/docs/vale/ValeAsciidocLint.java:160]: Doesn't this file name begin with a `.`?

```suggestion
        List<String> command = new ArrayList<>(List.of("--config=.vale.ini",
```

**Comment by gsmet** [comment]
This PR is all borked. I'm closing it.

**Comment by ebullient** [comment]
yea.. I have no idea what happened to it other than it sat for a long time. I'll fix it in a clean one.

**Pull request closed without merging**